internal/logic/logic_test.go

```diff
@@ -120,3 +131,456 @@ func Test_htmlPdfServiceLogic_Ping(t *testing.T) {
+				mockDatasource := mock.NewMockDataSource(mockCtrl)
+				mockDatasource.EXPECT().SaveFile("1", []byte("abc"), time.Duration(0)).Return(errors.New(""))
+				mockDatasource.EXPECT().GetFile("1").Return([]byte(""), nil)
+				rec := &htmlPdfServiceLogic{
+					dsSvc: mockDatasource,
+					htSvc: mockHtmlsvc,
+				}
+				return rec
+			},
+			validateFunc: func(x *respModel.Response) {
+				expected := respModel.Response{
+					Status:  http.StatusInternalServerError,
+					Message: codes.GetErr(codes.ErrFileStoreFail),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(x, &expected) {
+					t.Errorf("want %v got %v", expected, x)
+				}
+			},
+		},
+		{
+			name: "Failure:: Replace:: GetJsonFromHtml fail",
+			setupFunc: func() *htmlPdfServiceLogic {
+				mockHtmlsvc := mock.NewMockHtmlToPdf(mockCtrl)
+				mockHtmlsvc.EXPECT().GetJsonFromHtml([]byte("abc")).Return(nil, errors.New(""))
+				mockDatasource := mock.NewMockDataSource(mockCtrl)
+				mockDatasource.EXPECT().GetFile("1").Return([]byte(""), nil)
+				rec := &htmlPdfServiceLogic{
+					dsSvc: mockDatasource,
+					htSvc: mockHtmlsvc,
+				}
+				return rec
+			},
+			validateFunc: func(x *respModel.Response) {
+				expected := respModel.Response{
+					Status:  http.StatusInternalServerError,
+					Message: codes.GetErr(codes.ErrFileConversionFail),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(x, &expected) {
+					t.Errorf("want %v got %v", expected, x)
+				}
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rec := tt.setupFunc()
+			resp := rec.Replace("1", strings.NewReader("abc"))
+			tt.validateFunc(resp)
+		})
+	}
+
+}
+
+func Test_HtmlToPdf(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	tests := []struct {
+		name         string
+		requestBody  string
+		setupFunc    func() *htmlPdfServiceLogic
+		validateFunc func(*respModel.Response)
+	}{
+		{
+			name:        "Success:: HtmlToPdf",
+			requestBody: "1",
+			setupFunc: func() *htmlPdfServiceLogic {
+				//w := httptest.NewRecorder()
+				js, _ := json.Marshal(map[string]interface{}{
+					"Custom-Field": "hello",
+					"Pages": []interface{}{
+						"hello-world",
+						map[string]interface{}{
+							"Base64PageData": base64.StdEncoding.EncodeToString([]byte("abc")),
+							"Custom-Data":    "world",
+						},
+					},
+				})
+				mockHtmlsvc := mock.NewMockHtmlToPdf(mockCtrl)
+				mockHtmlsvc.EXPECT().GeneratePdf(gomock.Any(), js).Return(nil).Times(1)
+				mockDatasource := mock.NewMockDataSource(mockCtrl)
+				mockDatasource.EXPECT().GetFile("1").Return(js, nil) //PR COMMENT STILL IN PROGRESS
+				rec := &htmlPdfServiceLogic{
+					dsSvc: mockDatasource,
+					htSvc: mockHtmlsvc,
+				}
+				return rec
+			},
+			validateFunc: func(x *respModel.Response) {
+				expected := respModel.Response{
+					Status:  http.StatusOK,
+					Message: "SUCCESS",
+					Data:    nil,
+				}
+				if x != &expected {
```

Comment:
Struct comparison should be done using deep comparison.

Reply:
done